// stardust/dist/commonjs/elements/Input/Input.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.htmlInputPropNames = undefined;

var _includes2 = require('lodash/includes');

var _includes3 = _interopRequireDefault(_includes2);

var _pick2 = require('lodash/pick');

var _pick3 = _interopRequireDefault(_pick2);

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _lib = require('../../lib');

var _elements = require('../../elements');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var htmlInputPropNames = exports.htmlInputPropNames = [
// React
'selected', 'defaultValue', 'defaultChecked',

// Limited HTML props
'autoComplete', 'autoFocus', 'checked',
// 'disabled', do not pass (duplicates SUI CSS opacity rule)
'form', 'max', 'maxLength', 'min', 'name', 'pattern', 'placeholder', 'readOnly', 'required', 'step', 'type', 'value'];

/**
 * An Input is a field used to elicit a response from a user
 * @see Button
 * @see Form
 * @see Icon
 * @see Label
 */
function Input(props) {
  var action = props.action;
  var actionPosition = props.actionPosition;
  var children = props.children;
  var className = props.className;
  var disabled = props.disabled;
  var error = props.error;
  var focus = props.focus;
  var fluid = props.fluid;
  var icon = props.icon;
  var iconPosition = props.iconPosition;
  var inverted = props.inverted;
  var label = props.label;
  var labelPosition = props.labelPosition;
  var loading = props.loading;
  var size = props.size;
  var type = props.type;
  var input = props.input;
  var transparent = props.transparent;


  var classes = (0, _classnames2.default)('ui', size, (0, _lib.useValueAndKey)(actionPosition, 'action') || (0, _lib.useKeyOnly)(action, 'action'), (0, _lib.useKeyOnly)(disabled, 'disabled'), (0, _lib.useKeyOnly)(error, 'error'), (0, _lib.useKeyOnly)(focus, 'focus'), (0, _lib.useKeyOnly)(fluid, 'fluid'), (0, _lib.useKeyOnly)(inverted, 'inverted'), (0, _lib.useValueAndKey)(labelPosition, 'labeled') || (0, _lib.useKeyOnly)(label, 'labeled'), (0, _lib.useKeyOnly)(loading, 'loading'), (0, _lib.useKeyOnly)(transparent, 'transparent'), (0, _lib.useValueAndKey)(iconPosition, 'icon') || (0, _lib.useKeyOnly)(icon, 'icon'), className, 'input');

  var rest = (0, _lib.getUnhandledProps)(Input, props);
  var ElementType = (0, _lib.getElementType)(Input, props);
  var inputProps = (0, _pick3.default)(props, htmlInputPropNames);

  if (children) {
    return _react2.default.createElement(
      ElementType,
      _extends({}, rest, { className: classes }),
      children
    );
  }

  var actionElement = _elements.Button.create(action, function (elProps) {
    return {
      className: (0, _classnames2.default)(
      // all action components should have the button className
      !(0, _includes3.default)(elProps.className, 'button') && 'button')
    };
  });
  var iconElement = _elements.Icon.create(icon);
  var labelElement = _elements.Label.create(label, function (elProps) {
    return {
      className: (0, _classnames2.default)(
      // all label components should have the label className
      !(0, _includes3.default)(elProps.className, 'label') && 'label',
      // add 'left|right corner'
      (0, _includes3.default)(labelPosition, 'corner') && labelPosition)
    };
  });

  return _react2.default.createElement(
    ElementType,
    _extends({}, rest, { className: classes }),
    actionPosition === 'left' && actionElement,
    iconPosition === 'left' && iconElement,
    labelPosition !== 'right' && labelElement,
    (0, _lib.createHTMLInput)(input || type, inputProps),
    actionPosition !== 'left' && actionElement,
    iconPosition !== 'left' && iconElement,
    labelPosition === 'right' && labelElement
  );
}

Input._meta = {
  name: 'Input',
  type: _lib.META.TYPES.ELEMENT,
  props: {
    actionPosition: ['left'],
    iconPosition: ['left'],
    labelPosition: ['left', 'right', 'left corner', 'right corner'],
    size: _lib.SUI.SIZES
  }
};

Input.defaultProps = {
  type: 'text'
};

Input.propTypes = {
  /** An element type to render as (string or function). */
  as: _lib.customPropTypes.as,

  /** An Input can be formatted to alert the user to an action they may perform */
  action: _lib.customPropTypes.some([_react.PropTypes.bool, _lib.customPropTypes.every([_lib.customPropTypes.disallow(['children']), _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object, _react.PropTypes.element])])]),

  /** An action can appear along side an Input on the left or right */
  actionPosition: _react.PropTypes.oneOf(Input._meta.props.actionPosition),

  /** Primary content.  Used when there are multiple Labels or multiple Actions. */
  children: _lib.customPropTypes.every([_lib.customPropTypes.disallow(['input', 'label']), _lib.customPropTypes.givenProps({ action: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object, _react.PropTypes.element]).isRequired }, _lib.customPropTypes.disallow(['action'])), _lib.customPropTypes.givenProps({ icon: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object, _react.PropTypes.element]).isRequired }, _lib.customPropTypes.disallow(['icon'])), _react.PropTypes.node]),

  /** Additional classes. */
  className: _react.PropTypes.string,

  /** An Input field can show that it is disabled */
  disabled: _react.PropTypes.bool,

  /** An Input field can show the data contains errors */
  error: _react.PropTypes.bool,

  /** An Input field can show a user is currently interacting with it */
  focus: _react.PropTypes.bool,

  /** Take on the size of it's container */
  fluid: _react.PropTypes.bool,

  /** Optional Icon to display inside the Input */
  icon: _lib.customPropTypes.some([_react.PropTypes.bool, _lib.customPropTypes.every([_lib.customPropTypes.disallow(['children']), _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object, _react.PropTypes.element])])]),

  /** An Icon can appear inside an Input on the left or right */
  iconPosition: _react.PropTypes.oneOf(Input._meta.props.iconPosition),

  /** Format to appear on dark backgrounds */
  inverted: _react.PropTypes.bool,

  /** Shorthand prop for creating the HTML Input */
  input: _lib.customPropTypes.every([_lib.customPropTypes.disallow(['children']), _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object, _react.PropTypes.element])]),

  /** Optional Label to display along side the Input */
  label: _lib.customPropTypes.every([_lib.customPropTypes.disallow(['children']), _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object, _react.PropTypes.element])]),

  /** A Label can appear outside an Input on the left or right */
  labelPosition: _react.PropTypes.oneOf(Input._meta.props.labelPosition),

  /** An Icon Input field can show that it is currently loading data */
  loading: _react.PropTypes.bool,

  /** An Input can vary in size */
  size: _react.PropTypes.oneOf(Input._meta.props.size),

  /** Transparent Input has no background */
  transparent: _react.PropTypes.bool,

  /** The HTML input type */
  type: _react.PropTypes.string
};

exports.default = Input;
